lens.cabal

```diff
@@ -348,6 +348,8 @@ library
 
   if impl(ghc >= 8)
     ghc-options: -Wno-missing-pattern-synonym-signatures
+    ghc-options: -Wno-trustworthy-safe
```

Comment:
If we're going to enable this flag, we ought to guard it behind `if impl(ghc >= 7.10)`, since that's when this flag was introduced. Currently, we get warnings on GHC 7.10 as a result of not doing this ([example](https://travis-ci.org/ekmett/lens/jobs/639404566#L1055)).

Reply:
Done. Build `-Werror` clean when tested locally, on all GHC since 7.4